code/go/0chain.net/smartcontract/dbs/event/authorizer.go

```diff
@@ -86,15 +78,42 @@ func (a *Authorizer) exists(edb *EventDb) (bool, error) {
 
 	result := edb.Get().
 		Model(&Authorizer{}).
-		Where(&Authorizer{AuthorizerID: a.AuthorizerID}).
+		Where(&Authorizer{ID: a.ID}).
 		Count(&count)
 
 	if result.Error != nil {
 		return false,
 			fmt.Errorf(
 				"error searching for authorizer %v, error %v",
-				a.AuthorizerID, result.Error,
+				a.ID, result.Error,
 			)
 	}
 	return count > 0, nil
 }
+
+func NewUpdateAuthorizerTotalStakeEvent(ID string, totalStake currency.Coin) (tag EventTag, data interface{}) {
+	return TagUpdateAuthorizerTotalStake, Authorizer{
+		ID: ID,
+		Provider: &Provider{
+			TotalStake: totalStake,
+		},
+	}
+}
+
+func (edb *EventDb) updateAuthorizersTotalStakes(authorizer []Authorizer) error {
+	return edb.Store.Get().Clauses(clause.OnConflict{
+		Columns:   []clause.Column{{Name: "id"}},
+		DoUpdates: clause.AssignmentColumns([]string{"total_stake"}),
+	}).Create(&authorizer).Error
+}
+
+func mergeUpdateAuthorizerTotalStakesEvents() *eventsMergerImpl[Authorizer] {
+	return newEventsMerger[Authorizer](TagUpdateAuthorizerTotalStake, withAuthorizerTotalStakesAdded())
+}
+
+func withAuthorizerTotalStakesAdded() eventMergeMiddleware {
+	return withEventMerge(func(a, b *Authorizer) (*Authorizer, error) {
+		//use the newer one
```

Comment:
Please use `withUniqueEventOverwrite` for the mergeUpdateAuthorizerTotalStakesEvents if we don't need to do addition. So  as other providers.